open-banking-accelerator/components/com.wso2.openbanking.accelerator.identity/src/main/java/com/wso2/openbanking/accelerator/identity/app2app/validations/DigestValidator.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.wso2.openbanking.accelerator.identity.app2app.validations;
+
+import com.wso2.openbanking.accelerator.identity.app2app.model.DeviceVerificationToken;
+import com.wso2.openbanking.accelerator.identity.app2app.validations.annotations.ValidateDigest;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Base64;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+/**
+ * Validator class for validating digest of a device verification token.
```

Comment:
shall we also mention digest is expected to be the hash of the request object if present. 
And also let's mention this is added to bind the token to the RO.

Reply:
Resolved in [83e631f](https://github.com/wso2/financial-open-banking/pull/67/commits/83e631f5cf7f0e52f6460082da104e93ca98e07e).